### PR TITLE
Implement FilterOrder Command

### LINKED ISSUE #94: Implement FilterOrder Command

### PR DESCRIPTION
**Pull request by kaixuan477**
Close #94

### CODE REVIEW

File: src/main/java/seedu/sellsavvy/logic/commands/ordercommands/FilterOrderCommand.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        requireNonNull(model);
+
+        Person selectedPerson = model.getSelectedPerson().get();
+        if (selectedPerson == null) {
+            throw new CommandException(MESSAGE_ORDERLIST_DOES_NOT_EXIST);
+        }
+
+        selectedPerson.updateFilteredOrderList(predicate);
+        return new CommandResult(
+                String.format(Messages.MESSAGE_ORDERS_LISTED_OVERVIEW, selectedPerson.getFilteredOrderList().size()));
```
Comment (FooChao) [inline src/main/java/seedu/sellsavvy/logic/commands/ordercommands/FilterOrderCommand.java:47]: Maybe let user know what they listed is completed or pending?

**Review by FooChao — COMMENTED** [review]
Some small nitpicks don't really have to fix now.
But what u really need to fix is addressbookParserTest and adding user guide

File: src/main/java/seedu/sellsavvy/logic/commands/ordercommands/FilterOrderCommand.java (hunk 1 above)
Comment (kaixuan477) [reply src/main/java/seedu/sellsavvy/logic/commands/ordercommands/FilterOrderCommand.java:47]: FindPersonCommand also does not support query feedback. Should implement for that also?

File: src/main/java/seedu/sellsavvy/logic/commands/ordercommands/FilterOrderCommand.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ * Filters and lists all orders under a specified person from the displayed order list,
+ * with an order status that matches the argument keywords.
+ * Keyword matching is case insensitive.
+ */
+public class FilterOrderCommand extends Command {
+
+    public static final String COMMAND_WORD = "filterOrder";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Filters the displayed order list for orders with "
+            + "matching order status (case-insensitive) and displays them as a list with index numbers.\n"
```
[the remark this answers is not in the document]
Comment (FooChao) [reply src/main/java/seedu/sellsavvy/logic/commands/ordercommands/FilterOrderCommand.java:26]: oops

File: src/main/java/seedu/sellsavvy/logic/commands/ordercommands/FilterOrderCommand.java (hunk 1 above)
Comment (FooChao) [reply src/main/java/seedu/sellsavvy/logic/commands/ordercommands/FilterOrderCommand.java:47]: made sense.I didn't notice it for findPerson. I guess no need change now in that case we should either change both or none.

File: src/test/java/seedu/sellsavvy/logic/parser/FilterOrderCommandParserTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        assertParseFailure(parser, "1",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, FilterOrderCommand.MESSAGE_USAGE));
+
+        // Invalid status
+        assertParseFailure(parser, "invalid",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, FilterOrderCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_validArgs_returnsFilterOrderCommand() {
```
[the remark this answers is not in the document]
Comment (kaixuan477) [reply src/test/java/seedu/sellsavvy/logic/parser/FilterOrderCommandParserTest.java:35]: Updated in commit.

File: src/main/java/seedu/sellsavvy/logic/commands/ordercommands/FilterOrderCommand.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (kaixuan477) [reply src/main/java/seedu/sellsavvy/logic/commands/ordercommands/FilterOrderCommand.java]: Updated in commit.

File: src/main/java/seedu/sellsavvy/model/person/Person.java [hunk 4]
```diff
@@ -90,7 +96,30 @@ public boolean isSamePerson(Person otherPerson) {
      * Returns the order list as an unmodifiable {@code ObservableList}.
      */
     public ObservableList<Order> getOrderUnmodifiableObservableList() {
```
[the remark this answers is not in the document]
Comment (kaixuan477) [reply src/main/java/seedu/sellsavvy/model/person/Person.java]: Updated in commit.

**Review by FooChao — APPROVED** [review]
LGTM! maybe see if ky want to check.

File: src/main/java/seedu/sellsavvy/logic/commands/ordercommands/DeleteOrderCommand.java [hunk 5]
```diff
@@ -48,15 +49,15 @@ public CommandResult execute(Model model) throws CommandException {
         if (selectedPerson == null) {
             throw new CommandException(MESSAGE_ORDERLIST_DOES_NOT_EXIST);
         }
-        ObservableList<Order> unmodifiableOrderList = selectedPerson.getOrderUnmodifiableObservableList();
+        List<Order> lastShownOrderList = selectedPerson.getFilteredOrderList();
         OrderList orderList = selectedPerson.getOrderList();
 
         if (index.getZeroBased() >= orderList.size()) {
```
[the remark this answers is not in the document]
Comment (kaixuan477) [reply src/main/java/seedu/sellsavvy/logic/commands/ordercommands/DeleteOrderCommand.java]: Right, updated in commit.

File: src/main/java/seedu/sellsavvy/logic/commands/ordercommands/DeleteOrderCommand.java [hunk 6]
```diff
@@ -48,15 +49,15 @@ public CommandResult execute(Model model) throws CommandException {
         if (selectedPerson == null) {
             throw new CommandException(MESSAGE_ORDERLIST_DOES_NOT_EXIST);
         }
-        ObservableList<Order> unmodifiableOrderList = selectedPerson.getOrderUnmodifiableObservableList();
+        List<Order> lastShownOrderList = selectedPerson.getFilteredOrderList();
         OrderList orderList = selectedPerson.getOrderList();
```
[the remark this answers is not in the document]
Comment (kaixuan477) [reply src/main/java/seedu/sellsavvy/logic/commands/ordercommands/DeleteOrderCommand.java]: Right, updated in commit.

File: src/main/java/seedu/sellsavvy/model/order/OrderList.java [hunk 7]
```diff
@@ -66,15 +65,18 @@ public void add(Order toAdd) {
     }
 
     /**
-     * Removes the order of the corresponding index from the list.
-     * The index of order must exist in the list.
+     * Removes the first instance of an equivalent person from the list.
```
[the remark this answers is not in the document]
Comment (kaixuan477) [reply src/main/java/seedu/sellsavvy/model/order/OrderList.java]: Right, updated in commit.

**Review by awdse22 — APPROVED** [review]

File: src/test/java/seedu/sellsavvy/logic/commands/ordercommands/FilterOrderCommandTest.java [hunk 8]
```diff
@@ -0,0 +1,111 @@
...
+import seedu.sellsavvy.model.person.Person;
+
+/**
+ * Contains integration tests (interaction with the Model) for {@code FilterOrderCommand}.
+ */
+public class FilterOrderCommandTest {
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs()).createCopy();
+
+    @Test
+    public void equals() {
```
Comment (yu-sutong) [inline src/test/java/seedu/sellsavvy/logic/commands/ordercommands/FilterOrderCommandTest.java:35]: Very extensive testing. I like this.

File: docs/UserGuide.md [hunk 9]
```diff
@@ -206,6 +206,21 @@ Examples:
...
+### Filter orders under a Person by order status : `filterOrder`
+
+Filters orders by the specified order status, under a selected person from their displayed order list.
+
+Format: `filterOrder ORDER_STATUS`
+
+* A person's order list must first be displayed before filtering their order list.
+* Filters the selected person's order list for orders that match the `ORDER_STATUS`.
+* The order status **must be a valid status**. e.g. Completed, Pending.
+* `ORDER_STATUS` is case-insensitive. e.g. `completed` is the same as `Completed`.
```
Comment (yu-sutong) [inline docs/UserGuide.md:218]: Should we add another line that says that ORDER_STATUS can only be completed or pending?

**Review by yu-sutong — APPROVED** [review]
Minor nitpicks but LGTM.

**Review by FooChao — APPROVED** [review]
Hello a second approval to prompt you to merge why you reply to git faster than on telegram

File: docs/UserGuide.md (hunk 9 above)
Comment (kaixuan477) [reply docs/UserGuide.md:218]: Noted on this, I leave open for now in case we add more statuses.

**Pull request merged**